src/smc-hub/postgres/changefeed-query.ts

```diff
@@ -53,11 +53,11 @@ function key(obj: { [key: string]: any }): string {
 type State = "ready" | "closed";
 
 class ThrottledTableQueue extends EventEmitter {
-  private table: string;
-  private queue: { [key: string]: TableQuery } = {};
-  private db: PostgreSQL;
-  private process_timer: any;
-  private interval_ms: number;
+  private table?: string;
```

Comment:
No!  It would be better not to.delete these on.close than have to
put unnecessary null checks everywhere.
I'm -1 to this.  I only delete things in close for garbage collection
reasons.  There must be a better way than null.checks everywhere.
Maybe one assertion at the top of all functions?

Reply:
well, I don't know the best way forward. Regarding garbage collection, I don't understand why they are deleted. Is the database object being recreated all the time? A string like `project_id` or `table` is not even a complex instantiated object. 

So ok, maybe we should just get rid of most `delete ...` lines and keep those which are really in some way meaningful. I can certainly try that instead and see how far I get.